packages/trip-form/src/util.js

```diff
@@ -69,36 +116,88 @@ export function getTransitSubmodeOptions(icons, modes, selectedModes) {
   });
 }
 
-function getPrimaryModeOption(icons, selectedModes) {
+/**
+ * Returns big primary "Take Transit" choice.
+ * @param {*} ModeIcon The icon component for rendering.
+ * @param {*} selectedModes An array of string that lists the modes selected for a trip query.
+ */
+function getPrimaryModeOption(ModeIcon, selectedModes) {
   return {
     id: "TRANSIT",
     selected: selectedModes.some(isTransit) && selectedModes.includes("WALK"),
     showTitle: false,
     text: (
       <span>
-        <ModeIcon icons={icons} mode="TRANSIT" />
+        <ModeIcon mode="TRANSIT" />
         Take Transit
       </span>
     ),
     title: "Take Transit"
   };
 }
 
-function getTransitCombinedModeOptions(icons, modes, selectedModes) {
+/**
+ * Returns the transit + access mode combinations.
+ * @param {*} icons The icon set to use.
+ * @param {*} modes The available modes to choose from.
+ * @param {*} selectedModes An array of string that lists the modes selected for a trip query.
+ * @param {*} selectedCompanies The companies to show as selected.
+ * @param {*} supportedCompanies The supported companies for certain modes.
+ */
+function getTransitCombinedModeOptions(
+  ModeIcon,
+  modes,
+  selectedModes,
+  selectedCompanies,
+  supportedCompanies
+) {
   const { accessModes } = modes;
   const modesHaveTransit = selectedModes.some(isTransit);
 
   return (
     accessModes &&
     accessModes.map(modeObj => {
       const modeStr = getModeString(modeObj);
+      const modeCompany = modeObj.company
+        ? modeObj.company.toUpperCase()
+        : null;
+
+      const id = `TRANSIT+${modeStr}${
+        modeObj.company ? `+${modeObj.company}` : ""
+      }`;
```

Comment:
I think id would be easier to read if you put the result of the company ternary into a variable above:

```
const company = modeObj.company ? `+${modeObj.company}` : ""
const id = `TRANSIT+${modeStr}${company}`;
```